### ISSUE TITLE
direct path to the thumbnails

### ISSUE THREAD

**Issue by punkish** [zenodo/zenodo#2301]
I am trying to `fetch` image thumbnails from Zenodo… in order to do that without running into `CORS` restrictions, I have to use the API url. The thumbnails are very conveniently available at aliases such as `https://zenodo.org/record/4624222/thumb50` but that is not useful for use with the `fetch` API. How can I programmatically get the direct urls to the thumbnails?

**Comment by slint** [comment]
Indeed, these thumbnail endpoints aren't under `/api` which is configured to not have the `CORS` restrictions. We should either:

- [ ] Add these endpoints in the CORS configuration
- [ ] Move/add equivalent endpoints under `/api`

**Comment by punkish** [comment]
thanks… do whatever is easier/makes sense for you… for me, what is impt is to access the various thumbnails without CORS restriction. Background… I am (gingerly) exploring creating a progressive web app, maybe even a native app (eventually), that will offer offline capabilities. So I want to retrieve the thumbnails and cache them in IndexedDb

**Comment by punkish** [comment]
@slint, one thing to note… it is impt (because of the motivation I mentioned above) for the uri to be predictable programmatically. For example,

```
"thumb250":"https://zenodo.org/api/iiif/v2/4f5eff4d-0c3c-4421-b907-6c5a485df058:c7562b06-6793-4f0a-a4b0-241b47483e1b:oo_10203.jpg/full/250,/0/default.jpg",
"thumbs":{
        "10":"https://zenodo.org/record/919512/thumb10",
        "50":"https://zenodo.org/record/919512/thumb50",
        "100":"https://zenodo.org/record/919512/thumb100",
        "250":"https://zenodo.org/record/919512/thumb250",
        "750":"https://zenodo.org/record/919512/thumb750",
        "1200":"https://zenodo.org/record/919512/thumb1200",
```

**thumb250** has actually non-predictable *real* url as generated by `iiif`. The predictable urls are great because I can write a program that figures out the url based on *only* the record id (for example, 919512 above). So, as you go about making the thumbnail uris free of CORS restrictions, keep in mind that they remain predictable, not that non-predictable hash that is right now for the actual url. Thanks

**Comment by slint** [comment]
In the end, introducing a new endpoint under `/api` is actually more consistent with our current CORS practice, i.e. only `/api` endpoints get the CORS treatment, which makes sense for "external" apps pulling data from Zenodo.

So we're basically looking at something like this: `https://zenodo.org/api/records/919512/thumb50` for each size. I'm looking at our access logs and cannot see actually anyone using the `thumb{size}` endpoints from the past 2 months. I'm about to say we keep both around (but deprecate the old one with a redirect to the new one).

**Comment by punkish** [comment]
> So we're basically looking at something like this: `https://zenodo.org/api/records/919512/thumb50` for each size. 

I think I get what you are saying but I am not entirely sure. So, I will paraphrase it (and you can correct me if my understanding is wrong). You will make the following available under CORS

```
https://zenodo.org/api/records/919512/thumb10
https://zenodo.org/api/records/919512/thumb50
https://zenodo.org/api/records/919512/thumb100
https://zenodo.org/api/records/919512/thumb250
https://zenodo.org/api/records/919512/thumb750
https://zenodo.org/api/records/919512/thumb1200
```

The above works for me.

> I'm looking at our access logs and cannot see actually anyone using the `thumb{size}` endpoints from the past 2 months.

I do, and intend to use it more and more

**Comment by punkish** [comment]
> @punkish if you can handle the occasional 404, that's fine of course :)

yes, that is not a problem, and when these are available under the api, then it will be even easier

**Comment by abubelinha** [comment]
Somehow related to this ... as you talk about [unpredictability of IIIF urls](https://github.com/zenodo/zenodo/issues/2301#issuecomment-1064094463).
Is it possible to get those IIIF urls through api for the case of Zenodo depositions with more than ONE image? (#1935 & #2028)

**Comment by abubelinha** [comment]
Somehow related to this ... as you talk about [unpredictability of IIIF urls](https://github.com/zenodo/zenodo/issues/2301#issuecomment-1064094463).
@slint is it possible to get those IIIF urls through api for the case of Zenodo depositions with more than ONE image? ([#1935](https://github.com/zenodo/zenodo/issues/1935#issuecomment-1076305845) & [#2028](https://github.com/zenodo/zenodo/issues/2028#issuecomment-674903427))
Thanks
@abubelinha

**Comment by palmajoao** [comment]
Any reason this record  
https://zenodo.org/api/records/13759554/thumb50 
or 
https://zenodo.org/records/13759554/thumb50
get a 404 error?

Been working on a [records retrieval with the API ](https://digitaf.eu/policy-briefings/)query and adding the thumbnail and works great!. However I haven't figured out why in some records the thumbnail doesn't show up...

**Comment by palmajoao** [comment]
Should have dug a bit more... So as far as I got the thumbnails are implemented with the [iiif API protocol](https://iiif.io/api/image/3.0/) and therefore the thumbnails can be accessed controling the iiif syntax.

For example if I want the 200px width thumbnail of this record:
https://zenodo.org/records/13759554 
I would request this:
https://zenodo.org/api/iiif/record:13759554:DSC_1004_1.JPG/full/200,/0/default.png

**Comment by punkish** [comment]
yea, crummy URL, but it works
